slice-core/src/main/java/com/cognifide/slice/core/internal/provider/SliceModelProvider.java

```diff
@@ -229,17 +234,21 @@ public Object get(String className, Resource resource) throws ClassNotFoundExcep
 	}
 
 	private <T> T get(Key<T> key, ExecutionContextImpl executionItem) {
+		InjectionMonitoringContext monitoringContext = sliceStats.startMonitoring();
+		monitoringContext.setInjecteeClass(key.getTypeLiteral().getRawType());
```

Comment:
Consider replacing two lines above with one line:
InjectionMonitoringContext monitoringContext = sliceStats.startMonitoring(key.getTypeLiteral().getRawType());

(also, read the next comment)